sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -1876,3 +1930,22 @@ case class Decade(child: Expression) extends UnaryExpression with ImplicitCastIn
     defineCodeGen(ctx, ev, c => s"$dtu.getDecade($c)")
   }
 }
+
+case class Epoch(child: Expression, timeZoneId: Option[String] = None)
+    extends UnaryExpression with ImplicitCastInputTypes with TimeZoneAwareExpression {
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(TimestampType)
+  override def dataType: DataType = DecimalType(20, 6)
```

Comment:
@MaxGekk, Out of curiosity, why is it `DecimalType(20, 6)`?

Reply:
I tried `DoubleType` but I was loosing 1 microsecond in some tests. Regarding `precision` (20) and `scale` (6).  6 was taken to have microseconds, 20 to cover supported range `[1, 9999]` of years + additional numbers to cover negative years that are out of the range ( but it would be nice to cover them).

Reply:
Could you add a comment about the above before line 1938, @MaxGekk ?